### PR TITLE
Synchronize status logger's map accesses.

### PR DESCRIPTION
**Pull request by pmuetschard**
Bug: http://b/198709411

### CODE REVIEW

File: core/app/status/logger.go [hunk 1]
```diff
@@ -34,19 +35,26 @@ func RegisterLogger(progressUpdateFreq time.Duration) Unregister {
...
 type statusLogger struct {
-	lastProgressUpdate map[*Task]time.Time
-	progressUpdateFreq time.Duration
+	lastProgressUpdate     map[*Task]time.Time
+	lastProgressUpdateLock sync.Mutex
+	progressUpdateFreq     time.Duration
 }
 
 func (l statusLogger) OnTaskStart(ctx context.Context, t *Task) {
 	log.I(ctx, "%v Started", t)
```
Comment (purvisa-at-google-com) [inline core/app/status/logger.go:44]: Should this move inside the lock? Has a task really started if it's in a queue waiting to acquire a lock?

File: core/app/status/logger.go [hunk 2]
```diff
@@ -34,19 +35,26 @@ func RegisterLogger(progressUpdateFreq time.Duration) Unregister {
...
 	l.lastProgressUpdate[t] = time.Now()
+	l.lastProgressUpdateLock.Unlock()
 }
 
 func (l statusLogger) OnTaskProgress(ctx context.Context, t *Task) {
+	l.lastProgressUpdateLock.Lock()
 	if time.Since(l.lastProgressUpdate[t]) > l.progressUpdateFreq {
-		log.I(ctx, "%v (%v%%) %v", t, t.Completion(), t.TimeSinceStart())
 		l.lastProgressUpdate[t] = time.Now()
+		l.lastProgressUpdateLock.Unlock()
```
Comment (purvisa-at-google-com) [inline core/app/status/logger.go:54]: Might be safer to unlock once at the end of the function and have a single return point?

**Review by purvisa-at-google-com — APPROVED** [review]
Just nits, LGTM

File: core/app/status/logger.go (hunk 1 above)
Comment (pmuetschard) [reply core/app/status/logger.go:44]: Moved the log message to after the lock. Don't want to log while holding the lock.

File: core/app/status/logger.go (hunk 2 above)
Comment (pmuetschard) [reply core/app/status/logger.go:54]: Done.

**Pull request merged**